mpitests/test_apply.py

```diff
@@ -0,0 +1,1321 @@
+
+        # s1 should only contain 1 and -1
+        assert np.allclose(res**2, 1, atol=tol)
+
+        mean = np.mean(res)
+        expected = -(np.cos(varphi) * np.sin(phi) + np.sin(varphi) * np.cos(theta)) / np.sqrt(2)
+        assert np.allclose(mean, expected, atol=tol)
+
+        var = np.var(res)
+        expected = (
+            3
+            + np.cos(2 * phi) * np.cos(varphi) ** 2
+            - np.cos(2 * theta) * np.sin(varphi) ** 2
+            - 2 * np.cos(theta) * np.sin(phi) * np.sin(2 * varphi)
+        ) / 4
+        assert np.allclose(var, expected, atol=tol)
+
+
+class TestTensorVar:
+    """Test tensor variance measurements."""
+
+    @pytest.mark.parametrize("C_DTYPE", [np.complex128, np.complex64])
+    def test_paulix_pauliy(self, C_DTYPE, tol=TOL_STOCHASTIC):
+        """Test that a tensor product involving PauliX and PauliY works correctly"""
+        num_wires = 3
+        comm = MPI.COMM_WORLD
+
+        dev_gpumpi = qml.device(
+            "lightning.gpu", wires=num_wires, mpi=True, shots=1000, c_dtype=C_DTYPE
+        )
+
+        theta = 0.432
+        phi = 0.123
+        varphi = -0.543
+
+        @qml.qnode(dev_gpumpi)
+        def circuit():
+            qml.RX(theta, wires=[0])
+            qml.RX(phi, wires=[1])
+            qml.RX(varphi, wires=[2])
+            qml.CNOT(wires=[0, 1])
+            qml.CNOT(wires=[1, 2])
+            return qml.var(qml.PauliX(wires=[0]) @ qml.PauliY(wires=[2]))
+
+        res = circuit()
+
+        expected = (
+            8 * np.sin(theta) ** 2 * np.cos(2 * varphi) * np.sin(phi) ** 2
+            - np.cos(2 * (theta - phi))
+            - np.cos(2 * (theta + phi))
+            + 2 * np.cos(2 * theta)
+            + 2 * np.cos(2 * phi)
+            + 14
+        ) / 16
+        assert np.allclose(res, expected, atol=tol)
+
+    @pytest.mark.parametrize("C_DTYPE", [np.complex128, np.complex64])
+    def test_pauliz_hadamard(self, C_DTYPE, tol=TOL_STOCHASTIC):
+        """Test that a tensor product involving PauliZ and PauliY and hadamard works correctly"""
+        num_wires = 3
+        comm = MPI.COMM_WORLD
+        dev_gpumpi = qml.device(
+            "lightning.gpu", wires=num_wires, mpi=True, shots=1000, c_dtype=C_DTYPE
+        )
+
+        theta = 0.432
+        phi = 0.123
+        varphi = -0.543
+
+        @qml.qnode(dev_gpumpi)
+        def circuit():
+            qml.RX(theta, wires=[0])
+            qml.RX(phi, wires=[1])
+            qml.RX(varphi, wires=[2])
+            qml.CNOT(wires=[0, 1])
+            qml.CNOT(wires=[1, 2])
+            return qml.var(qml.PauliZ(wires=[0]) @ qml.Hadamard(wires=[1]) @ qml.PauliY(wires=[2]))
+
+        res = circuit()
+
+        expected = (
+            3
+            + np.cos(2 * phi) * np.cos(varphi) ** 2
+            - np.cos(2 * theta) * np.sin(varphi) ** 2
+            - 2 * np.cos(theta) * np.sin(phi) * np.sin(2 * varphi)
+        ) / 4
+        assert np.allclose(res, expected, atol=tol)
+
+
+"""
+class TestProbs:
```

Comment:
Should this be commented out?

Reply:
yes, those lines were moved to `test_porbs.py`